internal/pkg/sync/gcp.go

```diff
@@ -0,0 +1,108 @@
+package sync
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	"golang.org/x/oauth2/jwt"
+)
+
+const url = "http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/token"
+
+type Response struct {
+	AccessToken string         `json:"access_token"`
+	ExpiresIn   *time.Duration `json:"expires_in"`
+	TokenType   string         `json:"token_type"`
+}
+
+//
+func (l *location) isGCR() bool {
+	gcr := l.getGCR()
+	return gcr
+}
+
+//
+func (l *location) getGCR() (gcr bool) {
+	url := strings.Split(l.Registry, ".")
+	gcr = url[len(url)-2] == "gcr" && url[len(url)-1] == "io"
+	return
+}
+
+func isGCE() bool {
+	resp, err := http.Head(url)
+	if err != nil {
+		return false
+	}
+
+	if resp.Header["Metadata-Flavor"][0] == "Google" {
```

Comment:
check for existence of Metadata-Flavor header, and for array length before accessing; equality check can then be returned directly